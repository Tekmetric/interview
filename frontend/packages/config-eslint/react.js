module.exports = {
  extends: [
    '@vercel/style-guide/eslint/browser',
    '@vercel/style-guide/eslint/typescript',
    '@vercel/style-guide/eslint/react'
  ].map(require.resolve),
  parserOptions: {
    project: '**/tsconfig.json',
    sourceType: 'module'
  },
  globals: {
    JSX: true
  },
  settings: {
    'import/resolver': {
      typescript: {
        project: '**/tsconfig.json',
        sourceType: 'module'
      },
      node: {
        extensions: ['.mjs', '.js', '.jsx', '.ts', '.tsx']
      }
    }
  },
  ignorePatterns: ['node_modules/', 'dist/', '.eslintrc.js', '**/*.css'],
  // add rules configurations here
  rules: {
    'import/no-default-export': 'off',
    'import/no-relative-packages': 'off',
    'react/button-has-type': 'off',
    'no-restricted-imports': [
      'error',
      {
        patterns: [
          {
            group: ['packages/*'],
            message: "Please use '@tekmetric/*' imports instead"
          }
        ]
      }
    ],
    'import/order': [
      'error',
      {
        'newlines-between': 'always',
        groups: [
          ['builtin', 'external'],
          'unknown',
          ['parent', 'sibling', 'index']
        ],
        pathGroups: [
          {
            pattern: '~/**',
            group: 'unknown',
            position: 'after'
          }
        ]
      }
    ],
    'react/function-component-definition': [
      'error',
      {
        namedComponents: 'arrow-function',
        unnamedComponents: 'arrow-function'
      }
    ]
  },
  overrides: [
    {
      files: ['*.config.js'],
      env: {
        node: true
      }
    },
    {
      files: ['*.ts', '*.tsx'],
      rules: {
        'no-unused-vars': 'off',
        '@typescript-eslint/no-invalid-void-type': 'off',
        'react/jsx-no-leaked-render': 'off',
        '@typescript-eslint/no-unused-vars': [
          'error',
          {
            vars: 'all',
            args: 'after-used',
            ignoreRestSiblings: true,
            argsIgnorePattern: '^_',
            varsIgnorePattern: '^_'
          }
        ],
        '@typescript-eslint/no-misused-promises': [
          'error',
          {
            checksVoidReturn: false
          }
        ]
      }
    },
    {
      files: ['*.test.ts', '*.test.tsx'],
      env: {
        jest: true
      },
      rules: {
        '@typescript-eslint/no-unsafe-call': 'off',
        '@typescript-eslint/no-unsafe-member-access': 'off'
      }
    }
  ]
}
