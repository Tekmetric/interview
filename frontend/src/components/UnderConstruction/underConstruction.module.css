.container {
  position: relative;
  height: 100%;
}

.container__inner {
  height: auto;
  padding: var(--space-lg);
  left: 50%;
  position: absolute;
  top: 50%;
  transform: translate(-50%, -50%);
  text-align: center;
  border: 2px dashed var(--white-color-transparent);
  border-radius: var(--border-radius-md);
}

.container h2 {
  color: var(--white-color-transparent);
  font-size: var(--font-size-xl);
  margin-bottom: var(--space-md);
  text-transform: uppercase;
}

.container p {
  color: var(--white-color-transparent);
  padding: var(--space-md);
}

.container img {
  width: 100%;
  margin: 2rem 0;
  opacity: 0.5;
}

@media (min-width: 768px) {
}
