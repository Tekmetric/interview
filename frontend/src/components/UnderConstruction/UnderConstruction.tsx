import { FC } from 'react';
import { Link } from 'react-router-dom';
import styles from './underConstruction.module.css';
import UnderConstructionImage from '../../assets/images/under_construction.png';

const UnderConstruction: FC = () => {
  return (
    <div className={styles.container}>
      <div className={styles.container__inner}>
        <h2>under construction</h2>
        <p>
          We&apos;re busy tuning things up under the hood. This page isn&apos;t quite ready yet, but
          working hard to get it running smoothly. Check back soon!
        </p>
        <img src={UnderConstructionImage} alt="Under Construction" />
        <Link className="link-button" to="/">
          Take Me Home
        </Link>
      </div>
    </div>
  );
};

export default UnderConstruction;
