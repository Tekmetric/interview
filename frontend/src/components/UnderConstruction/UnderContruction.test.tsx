import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import UnderConstruction from './UnderConstruction';

describe('UnderConstruction', () => {
  it('renders correctly', () => {
    const { container } = render(
      <MemoryRouter>
        <UnderConstruction />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
