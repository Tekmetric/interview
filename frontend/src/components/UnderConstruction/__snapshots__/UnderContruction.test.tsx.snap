// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UnderConstruction > renders correctly 1`] = `
<div>
  <div
    class="_container_b5be04"
  >
    <div
      class="_container__inner_b5be04"
    >
      <h2>
        under construction
      </h2>
      <p>
        We're busy tuning things up under the hood. This page isn't quite ready yet, but we're working hard to get it running smoothly. Check back soon!
      </p>
      <img
        alt="Under Construction"
        src="/src/assets/images/under_construction.png"
      />
      <a
        class="link-button"
        data-discover="true"
        href="/"
      >
        Take Me Home
      </a>
    </div>
  </div>
</div>
`;
